src/main/webapp/cluster-view/cluster-view.controller.js

```diff
@@ -1,26 +1,170 @@
 'use strict';
 
 angular.module('managementConsole')
-    .controller('ClusterViewCtrl', [
+  .controller('ClusterViewCtrl', [
     '$scope',
     '$stateParams',
     '$state',
+    '$q',
     'modelController',
-    function ($scope, $stateParams, $state, modelController) {
-            if (!modelController.isAuthenticated()) {
-                $state.go('/logout');
+    function ($scope, $stateParams, $state, $q, modelController) {
+      if (!modelController.isAuthenticated()) {
```

Comment:
I am reviewing my old code here :), but this is not the best practice: here we first load the controller, view will probably partially render, then we check if user is authenticated and if not, we send him to logout. As a result, we sometimes get flashing of view that we were not supposed to see.

Better practice is not to even load controller if user is not authenticated! That is usually done by checking conditions in main module, by listening to '$stateChangeStart' event -> there we can check if state is public or private, and if it is public, we can check if user is authenticated: if not, we can return user to login, and if yes, we can forward him to wanted destination. I suggest creating a separate issue for this one. Also, this way there is no need to check authentication in each controller, so there is no code duplication.